osu.Game/Screens/Ranking/ScorePanelList.cs

```diff
@@ -114,36 +130,56 @@ public ScorePanel AddScore(ScoreInfo score, bool isNewLocalScore = false)
                 };
             });
 
-            flow.Add(panel.CreateTrackingContainer().With(d =>
+            var trackingContainer = panel.CreateTrackingContainer().With(d =>
             {
                 d.Anchor = Anchor.Centre;
                 d.Origin = Anchor.Centre;
-            }));
+                d.Hide();
+            });
+
+            flow.Add(trackingContainer);
 
             if (IsLoaded)
-            {
-                if (SelectedScore.Value == score)
-                {
-                    SelectedScore.TriggerChange();
-                }
-                else
-                {
-                    // We want the scroll position to remain relative to the expanded panel. When a new panel is added after the expanded panel, nothing needs to be done.
-                    // But when a panel is added before the expanded panel, we need to offset the scroll position by the width of the new panel.
-                    if (expandedPanel != null && flow.GetPanelIndex(score) < flow.GetPanelIndex(expandedPanel.Score))
-                    {
-                        // A somewhat hacky property is used here because we need to:
-                        // 1) Scroll after the scroll container's visible range is updated.
-                        // 2) Scroll before the scroll container's scroll position is updated.
-                        // Without this, we would have a 1-frame positioning error which looks very jarring.
-                        scroll.InstantScrollTarget = (scroll.InstantScrollTarget ?? scroll.Target) + ScorePanel.CONTRACTED_WIDTH + panel_spacing;
-                    }
-                }
-            }
+                displayScore(trackingContainer);
 
             return panel;
         }
 
+        private void displayScore(ScorePanelTrackingContainer trackingContainer)
+        {
+            if (!IsLoaded)
+                return;
+
+            var score = trackingContainer.Panel.Score;
+
+            // Calculating score can take a while in extreme scenarios, so only display scores after the process completes.
+            scoreManager.GetTotalScoreAsync(score)
```

Comment:
Just on an initial read, does the fact that this is done at a per panel level mean that the list could be reordered multiple times during a single update pass (ie. if the user changes the score display mode)?

Reply:
I guess that's just this method name being misleading. Okay to rename this method to `focusScore`?

Reply:
Score display mode should not/does not reorder scores. That was the reason for https://github.com/ppy/osu/pull/14598.

But if the order of scores has changed when compared to osu!stable, then yes, you will get panels inserted inbetween other panels. Fixing that would require the API to paginate by the standardised total score instead.

Reply:
In practice this looks to work pretty well, so gonna resolve this for now.